### PR TITLE
Improve directional highlight prompts

### PR DESCRIPTION
**Pull request by ifreund**
#### Summary
SUMMARY: Interface "Improve directional highlight prompts."

#### Purpose of change
This is just something that was bugging me, and I feel like the new behavior is a UX enhancement.  It started off with  just opening/closing of doors but the scope expanded to standardizing all such prompts.

#### Describe the solution
Check adjacent tiles for valid targets before prompting the player to select one.  If there are none, exit with a message.  If there is exactly one, use  that as the target automatically.  If there are more than one, prompt as per the current behavior.

#### Testing
Tested all combinations of opening/closing with 0/1/many adjacent valid targets.

### CODE REVIEW

**Comment by BevapDin** [comment]
There is an overload of `choose_adjacent_highlight` that has a parameter `auto_select_if_single`, which pretty much does this for you.

**Comment by ifreund** [comment]
Thanks for the tip @BevapDin. I tweaked that function slightly to exhibit the behavior I wanted and went ahead and applied the same changes to everything using it.  This should give the UI a bit more consistency and avoid unnecessary prompts.

**Comment by snipercup** [comment]
I'm testing it. It's something to get used to. I tried examining something because I switched tileset and wanted to know what something is, but it would not let me examine it, instead I get:
image

It works fine when there is a table next to me though:
image


Is this what you mean to implement? I can still know what is what using x:
image

If there was an option to exclude 'examine' action to this auto-select feature, I would probably enable it.

File: src/iuse.cpp [hunk 1]
```diff
@@ -3000,7 +3001,7 @@ int iuse::siphon( player *p, item *it, bool, const tripoint & )
     }
     if( found_more_than_one ) {
         cata::optional<tripoint> pnt_ = choose_adjacent_highlight(
-                                            _( "Siphon from where?" ), f, false, true );
+                                            _( "Siphon from where?" ), _( "There is nothing to siphon nearby" ), f, false, true );
```
Comment (AMurkin) [inline src/iuse.cpp]:
```suggestion
                                            _( "Siphon from where?" ), _( "There is nothing to siphon nearby." ), f, false, true );
```

File: src/iuse.cpp [hunk 2]
```diff
@@ -2738,7 +2737,8 @@ int iuse::dig( player *p, item *it, bool t, const tripoint & )
     };
 
     const cata::optional<tripoint> pnt_ = choose_adjacent_highlight(
-            _( "Deposit excavated materials where?" ), f, false );
+            _( "Deposit excavated materials where?" ),
+            _( "There is nowhere to deposit the excavated materials" ), f, false, true );
```
Comment (AMurkin) [inline src/iuse.cpp]:
```suggestion
            _( "There is nowhere to deposit the excavated materials." ), f, false, true );
```

**Comment by ifreund** [comment]
@snipercup Yes, that is what I indented to implement, though the new behavior can be reverted for just examine if need be.  This change pushes a little farther in the direction of separation of uses for the various actions.  My thoughts are that, as you point out, `x` or `;` to look around is the "proper" hotkey for checking what a tile is and the `e` should be used only if interaction with that tile is desired.

Regarding deconstruction and the scope of this PR, deconstruction doesn't currently use the `choose_adjacent_highlight()` function so migrating that over would be a little more involved and imo out of scope for this one.

**Comment by ifreund** [comment]
As @esotericist pointed out, auto select for examine isn't great if the player is next to one valid target and a second invalid target that they wish to check for interactivity.  I've kept the new failure message if there are no valid targets however.

**Comment by ifreund** [comment]
Quoting from the last commit message:

There are good arguments for and against this behavior, so the only
reasonable thing to do is to make it configurable.

Enabling this saves keypresses plain and simple.

Disabling this makes directional actions more consistent since one can
always count on pressing the action key and the direction in quick
succession without waiting to see if a target was autoselected.

The default was subjectively chosen to be true based on general
impressions of popularity.

The new behavior of bailing with a helpful prompt when there are no valid targets was not made configurable and stayed as is.

**Comment by wapcaplet** [comment]
I don't understand how any tile could be an invalid target for the "examine" action. Consider these messages as I enter a dark room and examine the area around me, before this change:

```
That is a chair.
That is a bed.
That is a rock floor.
That is a wall.
That is a dresser.
It is empty.
```

Now, I can only examine my surroundings like this if there are at least 2 tiles nearby containing items, doors, windows or other interactive things.

Although the "examine" keyboard shortcut has been repurposed to automatically do all kinds of different things, I think it's important to remember the original meaning of the word "examine" in such a strongly text-based role-playing environment. Thank you for making it configurable! I only wish the original behavior had been left as the default.

**Comment by ifreund** [comment]
@wapcaplet I personally turn this auto setting off as well and since this PR has brought consistency to all the directional actions I think it's an improvement once you find the setting.  Before, some actions auto selected a target like pry, and others didn't like examine.

I totally see where you're coming from with the use of examine to look at nearby terrain, but there's been a movement towards separation of functions between the different actions recently.  The look around hotkey, `;` or `x` by default, should serve just as well if not better for figuring out what the tiles around you represent.  Perhaps examine would be more aptly named "interact" these days

**Comment by wapcaplet** [comment]
Unfortunately, the "Autoselect if exactly one valid target" setting has no effect on the "examine" shortcomings I mentioned above. Adjacent structures, furniture, and terrain can no longer be examined at all, unless one of them happens to be interactive. #37745

**Pull request merged**